consensus/istanbul/config.go

```diff
@@ -181,3 +185,13 @@ func (c Config) GetConfig(blockNumber *big.Int) Config {
 	}
 	return newConfig
 }
+
+func (c Config) GetValidatorContractAddress(blockNumber *big.Int) common.Address {
+	validatorContractAddress := c.ValidatorContract
+	for i := 0; c.Transitions != nil && i < len(c.Transitions) && c.Transitions[i].Block.Cmp(blockNumber) <= 0; i++ {
```

Comment:
should we start from the end and break the loop at first occurence found?

Reply:
its kind of the same starting from the end or the start, it goes either way depending on the blockNumber